docs/development/release-process.rst

```diff
@@ -6,19 +6,14 @@ Release Process
 
     $ pip install nox
 
-#. Modify the ``CHANGELOG.rst`` to include changes made since the last release
```

Comment:
Don't we still need to add the entries?

Reply:
I mean, I guess? We don't have to call this out in the process tho -- we should really be doing it on a per-PR basis.

Anyway, we can re-add in a follow up.